### PR TITLE
support downloading empty folders

### PR DESCRIPTION
**Pull request by gykuma**
Add support to also download the empty folders in artifact ([issue](https://github.com/Microsoft/vsts-tasks/issues/7604))

### CODE REVIEW

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts [hunk 1]
```diff
@@ -128,8 +128,20 @@ export class ArtifactEngine {
                     return value;
                 });
```
Comment (omeshp) [inline Extensions/ArtifactEngine/Engine/artifactEngine.ts]: Add tests to cover this.

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts [hunk 2]
```diff
@@ -128,8 +128,20 @@ export class ArtifactEngine {
...
 
-                this.artifactItemStore.addItems(items);
-                this.artifactItemStore.updateState(item, models.TicketState.Processed);
+                if (items.length > 0) {
+                    this.artifactItemStore.addItems(items);
+                    this.artifactItemStore.updateState(item, models.TicketState.Processed);
+                }
+                else {
+                    destProvider.putArtifactItem(item, null)
```
Comment (omeshp) [inline Extensions/ArtifactEngine/Engine/artifactEngine.ts]: add a UT for artifact engine too?

Comment (omeshp) [inline Extensions/ArtifactEngine/Engine/artifactEngine.ts]: add a UT for artifact engine too?

**Review by omeshp — APPROVED** [review]

File: Extensions/ArtifactEngine/Engine/artifactEngine.ts (hunk 2 above)
Comment (gykuma) [reply Extensions/ArtifactEngine/Engine/artifactEngine.ts]: will take it up in next PR along with e2e tests

**Pull request merged**